### PR TITLE
More examples are moved to examples directory.

### LINKED ISSUE #14: [Documentation] Move Examples to a Directory
## What?
Currently, all examples are on [README.md][1] making scroll forever. Need to move them to a directory `examples/` with respective document names and hyperlinks, except for [installation][2] and [examples][3] (which only has 3 code blocks).

## Why?
[README.md][1] too lengthy.

## Additional Context
NA

## Ownership of Contribution
If the triage is approved,
- [ ] I wish to submit the PR myself.
- [x] I wish to submit the PR. But if someone is willing to do it, go ahead.
- [ ] I don't know how to fix this or make these changes. I wish others to pick this up for me.

[1]: https://github.com/vaguecoder/gorilla-mux/blob/master/README.md
[2]: https://github.com/vaguecoder/gorilla-mux/blob/master/README.md#install
[3]: https://github.com/vaguecoder/gorilla-mux/blob/master/README.md#examples

### PR DESCRIPTION
**Pull request by Rishav1707**
## This PR closes the following issue:
closes #14 

## Changes made:
- More examples are moved to the `examples/` directory.

### CODE REVIEW

**Comment by tanryberdi** [comment]
Hey @Rishav1707 , could you please add `Signed-off-by: Random J Developer <random@developer.example.org>` to end of your commit message

**Comment by tanryberdi** [comment]
> > Hey @Rishav1707 , could you please add `Signed-off-by: Random J Developer <random@developer.example.org>` to end of your commit message
> 
> But, how?

you can use `git commit --amend` to edit your commit message

**Comment by Rishav1707** [comment]
> Hey @Rishav1707 , could you please add `Signed-off-by: Random J Developer <random@developer.example.org>` to end of your commit message

Done @tanryberdi

**Comment by vaguecoder** [comment]
> > Hey @Rishav1707 , could you please add `Signed-off-by: Random J Developer <random@developer.example.org>` to end of your commit message
> 
> But, how?

Details of DCO are missing as #11 is still not in place. You need to add your name and email to git and use `git commit -s ...` while committing changes. Please refer to [this][1].

[1]: https://docs.pi-hole.net/guides/github/how-to-signoff/

File: README.md [hunk 1]
```diff
@@ -83,726 +83,7 @@ func ArticlesCategoryHandler(w http.ResponseWriter, r *http.Request) {
...
-func main() {
-    r := mux.NewRouter()
-    // Routes consist of a path and a handler function.
-    r.HandleFunc("/", YourHandler)
-
-    // Bind to a port and pass our router in
-    log.Fatal(http.ListenAndServe(":8000", r))
-}
-```
+And this is all you need to know about the basic usage. More advanced options are explained in [examples](./examples/examples.md).
```
Comment (vaguecoder) [inline README.md:86]: Not required. Hyperlinks from the following section in `README.md` should be linked to respective documents in examples.

```

* [Install](#install)
* [Examples](#examples)
* [Matching Routes](#matching-routes)
* [Static Files](#static-files)
* [Serving Single Page Applications](#serving-single-page-applications) (e.g. React, Vue, Ember.js, etc.)
* [Registered URLs](#registered-urls)
* [Walking Routes](#walking-routes)
* [Graceful Shutdown](#graceful-shutdown)
* [Middleware](#middleware)
* [Handling CORS Requests](#handling-cors-requests)
* [Testing Handlers](#testing-handlers)
* [Full Example](#full-example)
```

**Comment by vaguecoder** [comment]
@Rishav1707 As mentioned in #14, we need one sheet for each topic.
Say, [Matching Routes](https://github.com/vaguecoder/gorilla-mux#matching-routes) will be in `matching-routes.md`

**Comment by Rishav1707** [comment]
> @Rishav1707 As mentioned in #14, we need one sheet for each topic. Say, [Matching Routes](https://github.com/vaguecoder/gorilla-mux#matching-routes) will be in `matching-routes.md`

Done @vaguecoder

File: README.md [hunk 2]
```diff
@@ -83,726 +83,7 @@ func ArticlesCategoryHandler(w http.ResponseWriter, r *http.Request) {
...
-func main() {
-    r := mux.NewRouter()
-    // Routes consist of a path and a handler function.
-    r.HandleFunc("/", YourHandler)
-
-    // Bind to a port and pass our router in
-    log.Fatal(http.ListenAndServe(":8000", r))
-}
-```
+And this is all you need to know about the basic usage. More advanced options are explained in [examples](./examples/).
```
Comment (vaguecoder) [inline README.md:86]: Calling topics like static files, mw, etc. advanced topics is an overstatement. Let's remove this last line as hyperlinks are already in place. Rest all looks good.

**Comment by vaguecoder** [comment]
> > > Hey @Rishav1707 , could you please add `Signed-off-by: Random J Developer <random@developer.example.org>` to end of your commit message
> > 
> > 
> > But, how?
> 
> Details of DCO are missing as #11 is still not in place. You need to add your name and email to git and use `git commit -s ...` while committing changes. Please refer to [this](https://docs.pi-hole.net/guides/github/how-to-signoff/).

@Rishav1707 The DCO needs to be signed on all commits. Please squash all of your commits together using [ref.][1] and sign your commits using `-s` as we previously discussed. Ask in [discussion][2] section if you find it hard to follow.

Rest all looks good :thumbsup:

[1]: https://stackoverflow.com/a/5201642
[2]: https://github.com/vaguecoder/gorilla-mux/discussions/23

**Comment by vaguecoder** [comment]
As an extra commit was squashed together, there are conflicts. Please try as [discussed][1] and let us only post here once that is resolved.

[1]: https://github.com/vaguecoder/gorilla-mux/discussions/23#discussioncomment-4621393

**Review by vaguecoder — CHANGES_REQUESTED** [review]
Hold until squash is fixed.

**Comment by Rishav1707** [comment]
I am just closing this issue and committing another PR because it just messed up with different things
